app/views/users/edit.haml

```diff
@@ -8,15 +8,26 @@
         %th.subber{colspan: 2} Account
       %tr
         %th.sub= f.label :username
-        %td{class: cycle('even', 'odd')}= f.text_field :username, placeholder: "Username", disabled: current_user.salt_uuid.nil?
+        %td{class: cycle('even', 'odd')}
+          - disabled = current_user.salt_uuid.nil? && current_user.encrypted_password.nil?
+          = f.text_field :username, placeholder: "Username", disabled: disabled
+          - if disabled
+            %br
+            %span.details Please reset your password first to allow changing your username.
       %tr
         %th.sub= f.label :email
         %td{class: cycle('even', 'odd')}= f.text_field :email, placeholder: "Email address"
+      %tr
+        %th.sub Password
+        %td{class: cycle('even', 'odd')}
+          = link_to edit_user_registration_path do
+            %button{type: 'button'} Change Password
+      -# TODO: Disable my account #{button_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure?", turbo_confirm: "Are you sure?" }, method: :delete}
       %tr
         %th.sub Author profile
         %td{class: cycle('even', 'odd')}
-          %button{ type: 'button', onclick: "window.location.href='#{profile_edit_user_path(current_user)}'" }
-            Edit Profile
+          = link_to profile_edit_user_path(current_user) do
+            %button{type: 'button'} Edit Profile
```

Comment:
This could probably go in a different PR

Reply:
Moved to  #2372